app/scripts/controllers/mmi-controller.ts

```diff
@@ -853,26 +847,24 @@ export default class MMIController extends EventEmitter {
         internalAccount.id,
       );
     }
-    const selectedChainId = parseInt(
-      getCurrentChainId({ metamask: this.networkController.state }),
-      16,
-    );
-    if (selectedChainId !== chainId && chainId === 1) {
-      await this.networkController.setActiveNetwork('mainnet');
-    } else if (selectedChainId !== chainId) {
-      const { networkConfigurations } = this.networkController.state;
-
-      const foundNetworkConfiguration = Object.values(
-        networkConfigurations,
-      ).find(
-        (networkConfiguration) =>
-          parseInt(networkConfiguration.chainId, 16) === chainId,
-      );
 
-      if (foundNetworkConfiguration !== undefined) {
-        await this.networkController.setActiveNetwork(
-          foundNetworkConfiguration.id,
-        );
+    const selectedChainId = getCurrentChainId({
+      metamask: this.networkController.state,
+    });
+
+    if (selectedChainId !== toHex(chainId)) {
```

Comment:
Int the removed code there is a condition `&& chainId === 1`. Why is that condition not needed any more?

Reply:
The built in infura network configurations are now in state, where they were not before.  Therefore you don't need a special condition to call:

```ts
await this.networkController.setActiveNetwork('mainnet');
```

Since the infura/mainnet network is just a regular network whose network client id lives in networkConfigurationsByChainId

Reply:
@bergeron I can test this in mmi. Do you have the steps on how to test it manually?

Reply:
I'm not sure how to test MMI, or how it handles custom networks.  But the PR description describes the main flows under `Manual testing steps`.